python/ray/tune/suggest/bayesopt.py

```diff
@@ -111,23 +118,57 @@ def __init__(self,
             self._metric_op = 1.
         elif mode == "min":
             self._metric_op = -1.
+
         self._live_trial_mapping = {}
+        self._cached_results = []
+        self.random_search_trials = random_search_steps
+        self._total_random_search_trials = 0
 
         self.optimizer = byo.BayesianOptimization(
             f=None, pbounds=space, verbose=verbose, random_state=random_state)
 
         self.utility = byo.UtilityFunction(**utility_kwargs)
+
+        # Registering the provided analysis, if given
         if analysis is not None:
             self.register_analysis(analysis)
 
     def suggest(self, trial_id):
-        if self.max_concurrent:
-            if len(self._live_trial_mapping) >= self.max_concurrent:
+        """Return new point to be explored by black box function.
+
+        Parameters
+        --------------------
+        trial_id (str): Id of the trial.
+            This is a short alphanumerical string.
+
+        Returns
+        --------------------
+        Either a dictionary describing the new point to explore or
+        None, when no new point is to be explored for the time being.
+        """
```

Comment:
nit (standardization): Can you use the google docstring format? Thanks!

Reply:
Patched!